api/v1beta1/cryostat_types.go

```diff
@@ -469,6 +473,21 @@ type CertificateSecret struct {
 	CertificateKey *string `json:"certificateKey,omitempty"`
 }
 
+type Credential struct {
+	UserName        string
+	Password        string
+	MatchExpression string
+}
+
+type DeclarativeCredential struct {
+	// Name of secret in the local namespace
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
+	SecretName string `json:"secretName"`
+	// Key within secret containing the stored credential
+	// +optional
+	Credential string `json:"credential,omitempty"`
```

Comment:
I guess these should be added only to `v1beta2`, not `v1beta1`? @ebaron

Reply:
Right, new API should only go in v1beta2.